File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s = "".join(sorted(s))
        t = "".join(sorted(t))
        if len(s) == len(t):
            for i in range(0 , len(s)):
                if(s[i] != t[i]):
                    return False
            return True
        else:
            return False

        

File: 242-valid-anagram/valid-anagram.java
class Solution {
    // static {
    //     for (int i = 0; i < 500; i++) {
    //         isAnagram("a", "");
    //     }
    // }
    public static boolean isAnagram(String s, String t) {
        int slen = s.length();
        if (slen != t.length()) {
            return false;
        }

        int[] chars = new int[26];

        for (int i = 0; i < slen; i++) {
            chars[s.charAt(i) - 'a']++;
            chars[t.charAt(i) - 'a']--;
        }

        for (int count : chars) {
            if (count != 0) {
                return false;
            }
        }
        
        return true;
    }
}